src/components/TodoInput/TodoInput.tsx

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useContext, useEffect, useState } from 'react';
+import { USER_ID, createTodo, updateTodo } from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import classNames from 'classnames';
+import { Context } from '../Context/Context';
+
+type Props = {
+  setTempTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+  inputRef: React.RefObject<HTMLInputElement>;
+};
+
+export const TodoInput: React.FC<Props> = ({ setTempTodo, inputRef }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [title, setTitle] = useState('');
+  const { todos, setTodos, setErrorMessage } = useContext(Context);
+
+  const addTodo = (newTodo: Omit<Todo, 'id'>) => {
+    setIsLoading(true);
+
+    if (newTodo.title !== '') {
+      setTempTodo({ ...newTodo, id: 0 });
+
+      return createTodo(newTodo)
+        .then(createdTodo => {
+          setTodos(prev => [...prev, createdTodo]);
+          setTempTodo(null);
+          setTitle('');
+        })
+        .catch(() => {
+          setErrorMessage('Unable to add a todo');
+          setTempTodo(null);
+        })
+        .finally(() => {
+          setIsLoading(false);
+          if (inputRef.current) {
+            inputRef.current.focus();
+          }
+        });
+    } else {
+      setErrorMessage('Title should not be empty');
+    }
+
+    setIsLoading(false);
+
+    return null;
+  };
+
+  const todoToCreate = {
+    title: title.trim(),
+    userId: USER_ID,
+    completed: false,
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    addTodo(todoToCreate);
+  };
+
+  const switchCompleted = () => {
+    const val = todos.filter(t => t.completed).length !== todos.length;
+    const copyOfTodos = val ? todos.filter(t => !t.completed) : [...todos];
+
+    copyOfTodos.map(todo => {
```

Comment:
you don't need to make copy of array - it's redundant

Also, map is wrong method here - use forEach